book/src/qobject/generated-qobject.md

```diff
@@ -0,0 +1,112 @@
+<!--
+SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# `qobject::T` - The generated QObject
+
+One of the key features of CXX-Qt is the ability to create your own QObjects from Rust.
+This is what the [`#[cxx_qt::qobject]` macro](./qobject_struct.md) is for.
+This page serves to document the details of what is generated and how to interact with the generated QObject in a programmatic way.
+
+The `#[cxx_qt::qobject]` macro generates a qobject for a given Rust struct.
+Whilst this QObject is a C++ type, CXX-Qt will automatically wrap it as a [CXX Opaque Type](https://cxx.rs/extern-c++.html#opaque-c-types).
+These generated QObjects are accessible in a generated module with the name `qobject`.
+Therefore to access the generated QObject for a struct `T`, use `qobject::T`.
+
+## Anatomy
+
+Any QObject generated by CXX-Qt is just a C++ QObject subclass that owns an instance of the Rust struct.
+The instance of the Rust struct is constructed using its [`Default`](https://doc.rust-lang.org/std/default/macro.Default.html) implementation in the C++ constructor.
+Therefore implementing `Default` on the Rust struct is currently mandatory.
```

Comment:
```suggestion
Therefore the Rust struct must implement the `Default` trait.
```

"Currently"? I don't think we have any plans/desires to change this?

Reply:
I don't know whether we have an issue for this already.
But we could add a `constructor` trait that could be used instead of `Default`.
This trait could allow you to define types that require certain arguments to be constructed.

Reply:
Maybe, but for QML you need to have something that can be constructed without arguments anyway. This would only be useful for C++/widgets

Reply:
> Maybe, but for QML you need to have something that can be constructed without arguments anyway. This would only be useful for C++/widgets

Definitely.

If you guys want I can remove the "currently" part of this phrasing. Just wanted to keep that somewhat open.
We could also link to the discussion, to see if people are interested if they stumble over this.